crates/sui-json/src/lib.rs

```diff
@@ -293,29 +332,38 @@ fn is_homogeneous_rec(curr_q: &mut VecDeque<&JsonValue>) -> bool {
     // Process all in this queue/level
     while !curr_q.is_empty() {
         // Okay to unwrap since we know values exist
-        let curr = match curr_q.pop_front().unwrap() {
+        let v = curr_q.pop_front().unwrap();
+        let curr = match v {
             JsonValue::Bool(_) => ValidJsonType::Bool,
-            JsonValue::Number(_) => ValidJsonType::Number,
+            JsonValue::Number(x) if x.is_u64() => ValidJsonType::Number,
```

Comment:
we only support u64?

Reply:
JSON bid width is u64. That's the only way to check if numbers are unsigned.